src/Nethermind/Nethermind.Blockchain/Processing/ReadOnlyTransactionProcessorSource.cs

```diff
@@ -38,8 +38,9 @@ public ReadOnlyTransactionProcessorSource(ReadOnlyTxProcessingEnv environment)
         
         public IReadOnlyTransactionProcessor Get(Keccak stateRoot)
         {
+            var oldState = _environment.StateProvider.StateRoot;
             _environment.StateProvider.StateRoot = stateRoot;
```

Comment:
are you sure that the state provider has no changes here? (dirty state?) it would not be discovered in tests

Reply:
Yes looks good for now. Either this is needed or I can't share readOnlyEnvironment from block producer and need a separate per Constant contract call. This is lighter.